src/Compilers/Core/MSBuildTask/ManagedToolTask.cs

```diff
@@ -28,7 +28,16 @@ public abstract class ManagedToolTask : ToolTask
         /// </remarks>
         protected bool IsManagedTool => string.IsNullOrEmpty(ToolPath) && ToolExe == ToolName;
 
-        internal string PathToManagedTool => Utilities.GenerateFullPathToTool(ToolName);
+        /// <summary>
+        /// The directory where the tools (like csc) are located.
+        /// Can be an absolute path or a path relative to the build tasks assembly.
+        /// If <see langword="null"/>, the directory of the build tasks assembly is used on desktop runtime
+        /// and its <c>bincore</c> subdirectory is used on core runtime.
```

Comment:
This is pre-existing behavior right? i.e. the code path which now handles the `null` case, is just what we would always do before this method was introduced.